### PR TITLE
instcomp changes

### PR DESCRIPTION
**Pull request by ArcEye**
option extra_inst_cleanup now per instance via custom delete()
and passes all parameters to user function

Signed-off-by: Mick <arceye@mgware.co.uk>

### CODE REVIEW

**Comment by mhaberler** [comment]
uh oh, we need this one atop master, not halobject header - let me see if it does pick

**Comment by mhaberler** [comment]
yes it does!

all ok

**Comment by mhaberler** [comment]
sorry to be a pita - delete needs to pass back the return value of extra_inst_cleanup like so:
```c
static int delete(const char *name, void *inst, const int inst_size)
{
    return extra_inst_cleanup(name, inst, inst_size);
}
````
or just drop the delete() function altogether and 
```c
int rtapi_app_main(void)
{
   ...
    // without EXTRA_INST_CLEANUP
    comp_id = hal_xinit(TYPE_RT, 0, 0, instantiate, NULL, compname);
    // with EXTRA_INST_CLEANUP:
    comp_id = hal_xinit(TYPE_RT, 0, 0, instantiate, extra_inst_cleanup, compname);
   ...
}
```

your call

**Comment by mhaberler** [comment]
btw just squash the fix onto the previous and push-force so your name is in the commit

File: src/hal/utils/instcomp.g [hunk 1]
```diff
@@ -797,43 +800,45 @@ static int comp_id;
...
+#            strg = "    hal_print_msg(RTAPI_MSG_DBG,\"%s: int instance param: %s=%d\",__FUNCTION__,"
+#            strg += "\"%s\", %s);" % (to_c(name), to_c(name))
+#            print >>f, strg
+#        else:
+#            strg = "    hal_print_msg(RTAPI_MSG_DBG,\"%s: string instance param: %s=%s\",__FUNCTION__,"
+#            strg += "\"%s\", %s);" % (to_c(name), to_c(name))
+#            print >>f, strg
+#####################################################################################################################
+            
+        print >>f, "    return extra_inst_cleanup(name, inst, inst_size);"
```
Comment (ArcEye) [inline src/hal/utils/instcomp.g]: Did a soft reset and rewrote the original commit, as requested

**Comment by ArcEye** [comment]
Kept the delete() function, it may be useful in the future.
Won't add any overhead as only called when quitting anyway.

**Comment by mhaberler** [comment]
good choice! 

All works great! readying the examples now. Status is in https://github.com/mhaberler/machinekit/tree/pbring-instantiable 

pbring.c (C API) already warped and working

actually machinetalk/msgcomponents has a few .comp files which make great examples for instcomp with ringbuffers, will adapt those!

had to do a little detour, and bring in the nosetests fixes first so nosetests keep working; I deleted the runtests/nosetests commit for now until we see the light on RTAI (or dump RTAI for that matter ;)

**Pull request closed without merging**